extensions/amp-video/0.1/flexible-bitrate.js

```diff
@@ -289,3 +301,16 @@ function currentSource(video) {
     })
   );
 }
+
+/**
+ * @private
+ * @param {!Element} videoEl
+ * @return {number}
+ */
+function getBufferedPercentage(videoEl) {
+  let bufferedSum = 0;
+  for (let i = 0; i < videoEl.buffered.length; i++) {
+    bufferedSum += videoEl.buffered.end(i) - videoEl.buffered.start(i);
+  }
+  return bufferedSum / videoEl.duration;
```

Comment:
You should add an early return if readyState is 0, as `duration` would not be available

Reply:
SG, I think it's best to directly check for the duration to catch edge cases like `duration = 0` as well as `duration = null` by doing

```js
if (!videoEl.duration) {
  return 0;
}
```
at the start of the function

Reply:
Okay that's a good point. Please add a code comment to say that `duration` could be `NaN` (not available yet!) or `0`. `NaN` is always tricky to handle